src/main/java/org/jboss/tools/intellij/openshift/utils/odo/OdoCli.java

```diff
@@ -862,7 +867,12 @@ public String consoleURL() throws IOException {
 
     @Override
     public boolean isOpenShift() {
-        return ClusterHelper.getClusterInfo(client).isOpenshift();
+        OpenShiftClient osClient = client.adapt(OpenShiftClient.class);
+        try {
+            return osClient.isSupported();
+        } catch (KubernetesClientException e) {
+            return e.getCode() == HttpURLConnection.HTTP_UNAUTHORIZED;
```

Comment:
so if adapt to OpenshiftClient fails with HTTP_UNAUTHORIZED , it means it's an openshift cluster ?

Reply:
Yes, I have the same in intellij-kubernetes.